plugins/admin/core/root.go

```diff
@@ -0,0 +1,99 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"fmt"
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"knative.dev/client-contrib/plugins/admin/pkg"
+	"knative.dev/client-contrib/plugins/admin/pkg/command"
+	"knative.dev/client-contrib/plugins/admin/pkg/command/domain"
+	private_registry "knative.dev/client-contrib/plugins/admin/pkg/command/private-registry"
+	"os"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+
+func NewAdminCommand(params ...pkg.AdminParams) *cobra.Command {
+	p := &pkg.AdminParams{}
+	kubeConfig := os.Getenv("KUBECONFIG")
+	if kubeConfig == "" {
+		fmt.Println("cannot get cluster kube config, please export environment variable KUBECONFIG")
+		os.Exit(1)
+	}
+
+	cfg, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		fmt.Println("failed to build config:", err)
+		os.Exit(1)
+	}
+
+	p.ClientSet, err = kubernetes.NewForConfig(cfg)
+	if err != nil {
+		fmt.Println("failed to create client:", err)
+		os.Exit(1)
+	}
+
+	rootCmd := &cobra.Command{
+		Use:   "admin",
+		Short: "A plugin of kn client to manage Knative",
+		Long: `A plugin of kn client to manage Knative for administrators. 
+
+For example:
+kn admin domain set - to set Knative route domain
+kn admin private-registry enable - to enable deployment from the private registry
```

Comment:
Not sure why it "set" and "enable" for both being write operation. With "enable" I would expect also a "disable", but when we do have "set" then we could also set a "null" URL to indicate to remove a registry. Also, as mentioned I just would call it "registry".

Reply:
@rhuss 
As in above comment, I updated to `kn admin registry add`, I think this is more proper here, and later we can discuss more to add `registry list` or `registry rm`.